packages/block/src/block.ts

```diff
@@ -46,16 +46,19 @@ export class Block {
 
     // parse uncle headers
     const uncleHeaders = []
+    const uncleOpts: any = {
+      hardforkByBlockNumber: true,
+      ...opts, // This potentially overwrites hardforkByBlocknumber
+      // Use header common in case of hardforkByBlockNumber being activated
+      common: header._common,
+      // Disable this option here (all other options carried over), since this overwrites the provided Difficulty to an incorrect value
+      calcDifficultyFromHeader: undefined,
+    }
+    if (uncleOpts.hardforkByTD) {
+      delete uncleOpts.hardforkByBlockNumber
```

Comment:
I'm a little confused by:

- The initial setting of `hardforkByBlockNumber` as true; followed by
- The potential overwriting by `...options`; followed by
- The potential deletion when `uncleOpts.hardforkByTD` is `truthy`

Do we always want `uncleOpts.hardforkByTD` to take precedence over `hardforkByBlockNumber`?

Reply:
hardforkByTD is a superset of the old option, so that will always imply the old behavior (and add some new one).

That said: I didn't really understand this code in the first place, so my modest goal was just to preserve the existing functionality by doing it the most native and unfancy way. 😜